File: JAVA/oneBy2PowerN.java
/*
Compute exact value of 1/(2^n) for T cases
*/

import java.util.Scanner;
import java.math.BigInteger;

class oneBy2PowerN{
    public static void main(String args[]) {
		 Scanner in = new Scanner(System.in);
		 int T = in.nextInt(),len,num,i,j;
		 BigInteger b,five = BigInteger.valueOf(5);
		 String output;
		 for(i=0;i<T;i++){
			 num = in.nextInt();
			 b = five.pow(num);
			 len = num - b.toString().length();
			 output="0.";
			 for(j=0;j<len;j++)
			 output = output + "0";
			 System.out.println(output+b);
		 }
	  }
}

File: PYTHON/one-by-2-power-n.py
'''
Compute exact value of 1/(2^n) for T cases
'''
def oneBy2PowerN(N):
    b = pow(5,N)
    l = N - len(str(b))
    output = "0."
    for j in range(0, l):
        output = output + "0"
    return output + str(b)

print("Enter number of test cases: ",end="")
T = int(input())

for i in range(0, T):
    print("\nEnter the value of N: ",end="")
    num = int(input())
    print(oneBy2PowerN(num))
	
'''
With Regards,
Shivaji Varma
'''
